src/chainparams.cpp

```diff
@@ -338,28 +338,109 @@ class CRegTestParams : public CChainParams {
     }
 };
 
+/**
+ * Custom params for testing.
+ */
+class CCustomParams : public CChainParams {
+
+    void UpdateFromArgs(ArgsManager& args)
+    {
+        consensus.fPowAllowMinDifficultyBlocks = args.GetBoolArg("-con_fpowallowmindifficultyblocks", true);
+        consensus.fPowNoRetargeting = args.GetBoolArg("-con_fpownoretargeting", true);
+        consensus.nSubsidyHalvingInterval = args.GetArg("-con_nsubsidyhalvinginterval", 150);
+        consensus.BIP34Height = args.GetArg("-con_bip34height", 100000000);
+        consensus.BIP65Height = args.GetArg("-con_bip65height", 1351);
+        consensus.BIP66Height = args.GetArg("-con_bip66height", 1251);
+        consensus.nPowTargetTimespan = args.GetArg("-con_npowtargettimespan", 14 * 24 * 60 * 60); // two weeks
+        consensus.nPowTargetSpacing = args.GetArg("-con_npowtargetspacing", 10 * 60);
+        consensus.nRuleChangeActivationThreshold = args.GetArg("-con_nrulechangeactivationthreshold", 108); // 75% for testchains
+        consensus.nMinerConfirmationWindow = args.GetArg("-con_nminerconfirmationwindow", 144); // Faster than normal for custom (144 instead of 2016)
+        consensus.powLimit = uint256S(args.GetArg("-con_powlimit", "7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff"));
+        consensus.BIP34Hash = uint256S(args.GetArg("-con_bip34hash", "0x0"));
+        consensus.nMinimumChainWork = uint256S(args.GetArg("-con_nminimumchainwork", "0x0"));
+        // By default assume that the signatures in ancestors of this block are valid.
+        consensus.defaultAssumeValid = uint256S(args.GetArg("-con_defaultassumevalid", "0x00"));
+
+        nDefaultPort = args.GetArg("-ndefaultport", 18444);
+        nPruneAfterHeight = args.GetArg("-npruneafterheight", 1000);
+        fDefaultConsistencyChecks = args.GetBoolArg("-fdefaultconsistencychecks", true);
+        fRequireStandard = args.GetBoolArg("-frequirestandard", false);
+        fMineBlocksOnDemand = args.GetBoolArg("-fmineblocksondemand", true);
+        bech32_hrp = args.GetArg("-bech32_hrp", "bcrt");
+    }
+
+public:
+    CCustomParams(const std::string& chain, ArgsManager& args)
+    {
+        strNetworkID = chain;
+
+        consensus.vDeployments[Consensus::DEPLOYMENT_TESTDUMMY].bit = 28;
+        consensus.vDeployments[Consensus::DEPLOYMENT_TESTDUMMY].nStartTime = 0;
+        consensus.vDeployments[Consensus::DEPLOYMENT_TESTDUMMY].nTimeout = 999999999999ULL;
```

Comment:
Needs rebase, which raises the question why this class can not just derive from the CRegTestParams class...